hack/build-coredns.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Create a temporary directory for cloning openshift/coredns repo.
+# The directory will be deleted after the execution of script.
+BASE_PATH=$(mktemp -d)
+trap "chmod -R u+w $BASE_PATH; rm -rf $BASE_PATH" EXIT
+
+CURRENT_DIR=$(pwd)
+
+cd $BASE_PATH
+# Clone openshift/coredns repo if not already cloned.
+if [ ! -d coredns ]
+then
+    git clone https://github.com/openshift/coredns
+fi
+
+cd $CURRENT_DIR
+# Add the "ocp_dnsnameresolver" plugin to the cloned openshift/coredns repo.
+GO111MODULE=on GOFLAGS=-mod=mod $CURRENT_DIR/hack/add-plugin.sh $BASE_PATH/coredns
```

Comment:
How about passing the plugin path as the second argument to `hack/add-plugin.sh` to avoid the need to change the directory to this repository? This would avoid a lot of directory jumping, we could just create a tempdir, clone `coredns` into it and step into inside to do `add-plugin` and build `coredns`.

Reply:
Makes sense. I'll make the changes.